pyre/pyre-interpreter/src/module/_typing/_typing_app.py

```
@@ -415,72 +559,89 @@ def __repr__(self):
+
     def __init__(self, origin):
-        self.__origin__ = origin
+        object.__setattr__(self, "__origin__", origin)
 
     def __repr__(self):
-        return f"{self.__origin__.__name__}.args"
+        if type(self.__origin__) is ParamSpec:
+            return f"{self.__origin__.__name__}.args"
+        return f"{self.__origin__!r}.args"
 
     def __eq__(self, other):
```

Ruff (0.16.1)

[warning] Missing return type annotation for special method `__eq__`

(ANN204)